tests/unit/s2n_ktls_feature_probe_test.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "tls/s2n_ktls.h"
+
+#if defined(__linux__)
+    #include "linux/version.h"
+#endif
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* kTLS feature probe */
+    {
+#if defined(__linux__)
+    /* kTLS support was first added to AL2 starting in 5.10.130. */
+    #if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 130))
+        if (!platform_supports_ktls()) {
+            FAIL_MSG("kTLS feature probe is not working");
+        } else {
+            EXPECT_TRUE(true);
+        }
```

Comment:
Is this else necessary?

Reply:
If we dont have it then on a successful run it will print out "SKIPPED". We instead should print out "PASSED"

Reply:
Then you probably want to just do EXPECT_TRUE(s2n_ktls_is_supported_on_platform()) instead of the if/else. Is the concern that failure message won't be descriptive enough?

Reply:
I think we can live without the msg since we get the line number in the test output.